swarm/network/stream/stream.go

```diff
@@ -838,3 +838,27 @@ func (api *API) SubscribeStream(peerId enode.ID, s Stream, history *Range, prior
 func (api *API) UnsubscribeStream(peerId enode.ID, s Stream) error {
 	return api.streamer.Unsubscribe(peerId, s)
 }
+
+/*
+GetPeerSubscriptions is a API function which allows to query a peer for stream subscriptions it has.
+It can be called via RPC.
+It returns a map of node IDs with an array of string representations of Stream objects.
+*/
+func (api *API) GetPeerSubscriptions() map[string][]string {
```

Comment:
If this is used in tests only, how about having a separate test api object only loaded under certain conditions?

There's some code in the `pss` package using this approach, if you're curious.

Reply:
This is a brilliant and important question, and I am not clear about the correct answer. I would like to have a generaral test api object for sure.

The question is if this API function here is strictly for debugging or if we would like to have it available to query nodes always?

@nonsense @zelig

Reply:
Let's make a roundtable issue out of it and in any case a separate PR for it if a majority find this a good idea.

Reply:
@nolash can this be marked resolved for the time being?

Reply:
yes fine